packages/aes-kw/aes-kw.ts

```diff
@@ -0,0 +1,138 @@
+            throw new Error("aeskw: key data insufficient length must be a minimum of 128 bits or 16 bytes");
+        }
+
+        // Set A to the initial value
+        let A = this._iv;
+        // Initialize the length of the wrapped key, size always equals N+1
+        const wrappedKey = new Uint8Array(8*(N+1));
+        // Set the plain text into the wrapped key array offset by one
+        // 64 bit data block
+        wrappedKey.set(keyData, 8);
+
+        for (let j = 0; j < 6; j++) {
+            for (let i = 1; i <= N; i++) {
+                this._inputBuffer.set(A);
+                this._inputBuffer.set(wrappedKey.subarray(i*8,(i+1)*8),8);
+                this._cipher.encryptBlock(this._inputBuffer, this._outputBuffer);
+                A = writeUint64BE(i + j*N);
+                this.xor(A, this._outputBuffer.subarray(0,8));
+                wrappedKey.set(this._outputBuffer.subarray(8,16), i*8);
+            }
+        }
+
+        wrappedKey.set(A);
+        return wrappedKey;
+    }
+
+    /**
+     * Un-wraps a wrapped key using the key encryption key supplied in the
+     * constructor
+     * @param wrappedKey: The wrapped key to un-wrap with the key encryption key
+     */
+    unwrapKey(wrappedKey: Uint8Array): Uint8Array {
+        // Floor divide the length of the unwrapped key to determine 
+        // how many 64 bit data blocks it contains, N represents the number
+        // of 64 bit data blocks of the plain text which is always one less
+        // than the cipher text
+        const N = ((wrappedKey.length - (wrappedKey.length % 8)) / 8) - 1;
+        
+        // The keyData must be at minimum 128 bit (16 bytes) in length
+        if (N <= 1) {
+            throw new Error("aeskw: key data insufficient length must be a minimum of 128 bits or 16 bytes");
+        }
+
+        // Set A to the first 64 bit data block of the wrapped key
+        let A = wrappedKey.subarray(0,8);
+        // Initialize the length of the key data, size always equals N
+        const keyData = new Uint8Array(8*(N));
+
+        for (let j = 5; j >= 0; j--) {
+            for (let i = N; i >= 1; i--) {
+                this.xor(A, writeUint64BE(i + j*N));
+                this._inputBuffer.set(A);
+                this._inputBuffer.set(wrappedKey.subarray((i)*8,(i+1)*8),8);
+                this._cipher.decryptBlock(this._inputBuffer, this._outputBuffer);
+                A.set(this._outputBuffer.subarray(0,8));
+                wrappedKey.set(this._outputBuffer.subarray(8,16), i*8);
+            }
+        }
+
+        // Integrity check, the A component of the un-wrapped key buffer should
+        // equal the default initial value
+        if (compare(A, this._iv) == 0) {
+            throw new Error("aeskw: unwrapped data does not contain the default iv");
+        }
+
+        keyData.set(wrappedKey.subarray(8))
+        return keyData;
+    }
+
+    /**
+     * Internal method for bitwise XOR of two Uint8Arrays
```

Comment:
Could you replace "XOR of two Uint8Arrays" with "XOR of b into a" so that it's clear where the destination is? Actually, maybe just rename a to dst and b to src.

Also, this can be an unexported function in the package instead of the method.

Reply:
Thanks will do